payment/app.py

```diff
@@ -12,39 +14,174 @@
                               password=os.environ['REDIS_PASSWORD'],
                               db=int(os.environ['REDIS_DB']))
 
-
 def close_db_connection():
     db.close()
 
+order_url = os.environ['GATEWAY_URL'] + "/orders/"
+stock_url = os.environ['GATEWAY_URL'] + "/stock/"
+
+RANDOM_SEED = 444
+ID_BYTES_SIZE = 32
+random.seed(RANDOM_SEED)
 
 atexit.register(close_db_connection)
 
+class User:
+    def __init__(self, user_id: str):
+        self.user_id = user_id
+        self.credit = 0
+    
+    def to_dict(self):
+        return {
+            "user_id" : self.user_id,
+            "credit" : self.credit,
+        }
 
 @app.post('/create_user')
 def create_user():
-    pass
+
+    # Create new user id
+    new_user_id = f"user:{random.getrandbits(ID_BYTES_SIZE)}"
+    while db.hget(new_user_id, "user_id"):
+        new_user_id = f"user:{random.getrandbits(ID_BYTES_SIZE)}"
+    
+    new_user = User(user_id=new_user_id)
+    
+    try:
+        db.hset(new_user.user_id, mapping=new_user.to_dict())
+    except Exception as exp:
+        return Response(str(exp), status=400)
+
+    return_user = {
+        "user_id" : new_user_id,
+        "credit" : 0,
+    }
+
```

Comment:
Let's keep only the ```user_id``` on return - for consistency with the assignment and the other services.